packages/actor-init-sparql/lib/ActorInitSparql-browser.ts

```diff
@@ -31,6 +36,49 @@ export class ActorInitSparql extends ActorInit implements IActorInitSparqlArgs {
     super(args);
   }
 
+  /**
+   * Create a copy of the given operation in which all given bindings are applied.
+   * The bindings are applied to all quad patterns and path expressions.
+   *
+   * @param {Operation} operation An operation.
+   * @param {Bindings} initialBindings Bindings to apply.
+   * @return {Operation} A copy of the given operation where all given bindings are applied.
+   */
+  public static applyInitialBindings(operation: Algebra.Operation, initialBindings: Bindings): Algebra.Operation {
+    const copiedOperation: Algebra.Operation = <any> {};
+    for (const key of Object.keys(operation)) {
+      if (Array.isArray(operation[key])) {
+        if (key === 'variables') {
+          copiedOperation[key] = operation[key].filter(
+            (variable: RDF.Variable) => initialBindings.keySeq().indexOf(termToString(variable)) < 0);
+        } else {
+          copiedOperation[key] = operation[key].map(
+            (subOperation: Algebra.Operation) => ActorInitSparql.applyInitialBindings(subOperation, initialBindings));
+        }
+      } else if (ActorInitSparql.ALGEBRA_TYPES.indexOf(operation[key].type) >= 0) {
+        copiedOperation[key] = ActorInitSparql.applyInitialBindings(operation[key], initialBindings);
+      } else {
+        copiedOperation[key] = operation[key];
+      }
+
+      if (operation.type === 'pattern' || operation.type === 'path') {
```

Comment:
You should totally use Algebra.types.PATTERN and .PATH to make it look cooler (and more future proof) ;)